### PR TITLE
[RPA-1330] Display nested subprocesses in History table

### PR DESCRIPTION
**Pull request by seul88**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

---

I agree to the terms of the RunBotics Contributor License Agreement.

### CODE REVIEW

File: runbotics/runbotics-desktop/src/action/general/general.action-handler.ts [hunk 1]
```diff
@@ -58,16 +58,21 @@ export default class GeneralActionHandler extends StatelessActionHandler {
             if (processSystem !== BotSystem.ANY && processSystem !== system) {
                 reject(new Error(`Process with system (${processSystem}) cannot be run by the bot with system (${system})`));
             }
+                
+            const processInstancesIds = Object.keys(this.runtimeService.processInstances);
+            const parentProcessId = processInstancesIds[processInstancesIds.length - 1];
```
Comment (star-in-the-jar) [inline runbotics/runbotics-desktop/src/action/general/general.action-handler.ts]: what if processInstancesIds is empty?

Comment (star-in-the-jar) [reply runbotics/runbotics-desktop/src/action/general/general.action-handler.ts]: ah, I see it lower, but imo let's do it (parentProcessId ?? request.rootProcessInstanceId,) here

File: runbotics/runbotics-orchestrator-ui/src/src-app/components/tables/Table/Table.tsx [hunk 2]
```diff
@@ -48,9 +48,31 @@ const Table = <T extends object>({
...
+            return tableData.map(item => replaceKeyRecursive(item, newKey));
+        }
+      
+        const newObj = {};
+      
+        for (const key in tableData) {
+            if (Object.hasOwnProperty.call(tableData, key)) {
+                const newKeyString = key === subRowProperty ? newKey : key;
+                newObj[newKeyString] = replaceKeyRecursive(tableData[key] as T[], newKey);
+            }
```
Comment (star-in-the-jar) [inline runbotics/runbotics-orchestrator-ui/src/src-app/components/tables/Table/Table.tsx:66]: what does it do? can we make it more readable for future devs?

**Review by star-in-the-jar — APPROVED** [review]

File: runbotics/runbotics-orchestrator-ui/src/src-app/components/tables/Table/Table.tsx (hunk 2 above)
Comment (seul88) [reply runbotics/runbotics-orchestrator-ui/src/src-app/components/tables/Table/Table.tsx:66]: So the recursive function ProcessInstance.utils.ts file creates the tree structure where nested objects are stored as 'subprocesses' arrays. In this recursive function we want to bind those subprocesses to nested rows of table. They are organized in correct structure already, but we need to rename its' keys from value passed as subRowProperty props (in that case -'subprocesses') to 'subRows'.

**Pull request merged**